NthKey/Model/DataManager.swift

```diff
@@ -182,29 +183,38 @@ extension DataManager {
                 return
             }
 
-            var receivedNetwork: Network = .testnet
-            var ourHexString: String = ""
-            guard desc.extendedKeys.contains(where: { (key) -> Bool in
-                var result = false
-                for (networkKey, fingerprint) in fingerprints {
-                    guard key.fingerprint == fingerprint.hexString,
-                          let net = Network.valueFromStringKey(networkKey) else { continue }
-                    result = true
-                    receivedNetwork = net // TODO: Here we select network based on checking it's fingerprint. In case of fingrprints are the same then move network selection completely to the "Add wallet" flow.
-                    ourHexString = fingerprint.hexString
+            // Checking co-signers network
+            var receivedNetwork: Network?
+            for key in desc.extendedKeys {
+                var keyNetwork: Network?
+                switch key.xpub.prefix(4) {
```

Comment:
Instead of parsing the string like this, I think you can use the HDKey constructor in LibWally with the full string and see which network it returns?

Reply:
This initiliazer:
https://github.com/blockchain/libwally-swift/blob/master/LibWally/BIP32.swift#L157

And then HDKey class has a method which uses libwally-core internals to figure out the network: https://github.com/blockchain/libwally-swift/blob/master/LibWally/BIP32.swift#L213

Reply:
@Sjors Yes, you are right. fixed in 8d06e2524b6ba027f0eedbb9b17b3ac547ba6087